hist/hist/src/THLimitsFinder.cxx

```diff
@@ -381,8 +383,8 @@ void THLimitsFinder::OptimizeLimits(Int_t nbins, Int_t &newbins, Double_t &xmin,
       xmin = -1;
       xmax = 1;
    } else {
-      xmin    = binlow;
-      xmax    = binhigh;
+      xmin = std::min(binlow, xmin);
+      xmax = std::max(binhigh, xmax + 0.01 * (xmax - xmin));
```

Comment:
```suggestion
      xmax = std::max(binhigh, xmax + 1e-12 * (xmax - xmin));
```
Does it only work with 1% ?

Reply:
agreed, but it would be weird that there is a right margin but not a left margin? (That's why I was thinking of 1e-12 or 1e-6)
Anyway, both options work fine.